pvlib/location.py

```diff
@@ -124,7 +124,42 @@ def from_tmy(cls, tmy_metadata, tmy_data=None, **kwargs):
 
         # not sure if this should be assigned regardless of input.
         if tmy_data is not None:
-            new_object.tmy_data = tmy_data
+            new_object.weather_data = tmy_data
+
+        return new_object
+
+    @classmethod
+    def from_epw(cls, epw_metadata, epw_data=None, **kwargs):
```

Comment:
In retrospect, I wish the `from_tmy` method signature used `metadata` and `data` instead of `tmy_metadata` and `tmy_data`. Then both methods would have the signature `cls, metadata, data, **kwargs`. Adding `tmy_` or `epw_` to the argument names is redundant and makes it harder to reuse.